src/sentry/api/endpoints/release_thresholds/release_threshold_status_index.py

```diff
@@ -126,61 +161,161 @@ def get(self, request: Request, organization: Organization | RpcOrganization) ->
             .order_by("-date")
             .distinct()
         )
-        queryset.prefetch_related("projects__release_thresholds")
+        queryset.prefetch_related(
+            "projects__release_thresholds"
+        )  # maybe prefetch "deploy_set" as well?
 
         # ========================================================================
-        # TODO:
-        # Fetch relevant snuba/sessions data
-        # Determine which thresholds have succeeded/failed
-
-        release_threshold_health: DefaultDict[int, Any] = defaultdict()
+        # Step 3: flatten thresholds and compile projects/releases/thresholds by type
+        # ========================================================================
+        thresholds_by_type: DefaultDict[int, dict[str, list]] = defaultdict()
         for release in queryset:
-            release_project = defaultdict(list)
+            # TODO:
+            # We should update release model to preserve threshold states.
+            # if release.failed_thresholds/passed_thresholds exists - then skip calculating and just return thresholds
             if project_ids_list:
                 project_list = release.projects.filter(id__in=project_ids_list)
             else:
                 project_list = release.projects.all()
             for project in project_list:
-                project_threshold_statuses = []
                 if environments_list:
-                    thresholds_list = project.release_thresholds.filter(
+                    thresholds_list: List[ReleaseThreshold] = project.release_thresholds.filter(
                         environment__name__in=environments_list
                     )
                 else:
                     thresholds_list = project.release_thresholds.all()
                 for threshold in thresholds_list:
-                    is_healthy = self.is_threshold_healthy(threshold)
-                    project_threshold_statuses.append(
-                        {
-                            **serialize(threshold),
-                            "is_healthy": is_healthy,
+                    if threshold.threshold_type not in thresholds_by_type:
+                        thresholds_by_type[threshold.threshold_type] = {
+                            "projects": [],
+                            "releases": [],
+                            "thresholds": [],
                         }
+                    thresholds_by_type[threshold.threshold_type]["projects"].append(project.id)
+                    thresholds_by_type[threshold.threshold_type]["releases"].append(release.version)
+                    enriched_threshold: EnrichedThreshold = {
+                        # see https://github.com/python/mypy/issues/4122
```

Comment:
This issue is fixed. No?

Reply:
hmmm fair point.
Looks like the issue with mypy here may actually be due to our `serialize` helper not having an explicit return type 🤦 

I'll cast it 👍

Reply:
whelp. Looks like nothing likes the **splat method still so i'll just utilize `.update` instead of splatting values